pages/learn/deploy/offline-updates.md

```diff
@@ -0,0 +1,276 @@
+
+### Update Device Registration(s)
+
+This section deals with device registration in the cloud (balenaCloud or openBalena). The following steps help in manually reflecting the new state of the device for offline devices or devices on private networks without Internet access.
+
+#### Patch State Endpoint - Update Device State in the Cloud
+
+This step updates the device state in the cloud (balenaCloud or openBalena)
+on behalf of the device, as if the device was online. It is only required for
+devices that do not have an internet connection. If the device has an internet
+connection, even if low bandwidth, this step may be skipped because the device
+itself will be able to contact the cloud to update its state.
+
+```bash
+$ os_version_tag=$(echo ${os_version} | awk -F'+' '{print $1}')
+$ os_version_semver=$(echo ${os_version_tag} | sed 's/v//g')
+$ os_revision=$(echo ${os_version} | awk -F'+' '{print $2}' | awk -F'.' '{print $1}')
+$ os_variant=$(echo ${os_version} | awk -F'+' '{print $2}' | awk -F'.' '{print $2}')
+
+$ supervisor_version=$(curl --silent https://raw.githubusercontent.com/balena-os/meta-balena/${os_version_tag}/meta-balena-common/recipes-containers/resin-supervisor/resin-supervisor.inc \
+    | grep SUPERVISOR_TAG \
+    | awk '{print $3}' \
+    | sed 's/"//g' \
+    | sed 's/v//g')
+
+# (e.g.) staging
+$ release_id=$(curl --silent \
+    "https://api.balena-staging.com/v6/release?\$filter=startswith(commit,'${commit}')&\$select=id" \
+    -H "Authorization: Bearer $(cat ~/.balena/token)" | jq -r .d[].id)
+
+$ curl --silent \
+    -X 'PATCH' "https://api.balena-staging.com/v6/device(uuid='${uuid}')" \
+    -H "Authorization: Bearer $(cat ~/.balena/token)" \
+    -H 'Content-type: application/json' \
+    --data-binary "{\"os_variant\":\"${os_variant}\",\"os_version\":\"balenaOS ${os_version_semver}+${os_revision}\",\"supervisor_version\":\"${supervisor_version}\",\"is_running__release\":${release_id}}"
+```
+
+#### [Optional] Set Tags on Devices
+
+Tags help identify and track devices in your fleet as to what commit and OS version have they been flashed & updated offline with. One can use [`balena tag set`][balena-tag-set] to create new tags as needed.
+
+```bash
+$ balena tag set 'offline:commit' "${commit}" \
+  --device ${uuid}
+
+$ balena tag set 'offline:hostOS' "${os_version}" \
+  --device ${uuid}
+```
+
+Read more about how the process can work better for your usecase in the [offline updates](https://balena.io/blog/offline-updates-make-it-easier-to-update-balena-devices-without-the-internet) blog.
+Here's an [example project](https://github.com/balena-io-example/offline-updates/) to demonstrate offline updates for offline devices with user data mounted on external storage media persisting between hostOS and application updates.
```

Comment:
@vipulgupta2048 this "example project" doesn't exist. Is the intention here to point somewhere? Just to note that we can't point to https://github.com/balena-io-playground/offline-updates directly, so it would have to be moved to some other relevant GH org, where all of our docs refer to examples. If we are going to move this, we would need to refactor the README, since the content would already be in this doc.

Reply:
@ab77  Totally upto you, folks regarding the same. As Alex M and Andrew has mentioned before. Any example or project being referenced into the docs or blogs have to be in the `github.com/balena-io-examples` repo. Atleast that's the plan what I read in the blog as well. I do agree on keeping one source of truth and the documentation could be that. Easier to maintain for everyone.